### PR TITLE
Topic 3 Include download links to code

### PR DESCRIPTION
**Pull request by jameshughes89**
They can now download the Friend.java and ContactList.java files. 

Sphinx makes this easy with a :download: 'role'

### CODE REVIEW

File: source/topic3.rst [hunk 1]
```diff
@@ -768,6 +768,25 @@ Friend Count
...
     }
 
+.. code-block:: java
+    :linenos:
+    :emphasize-lines: 4
+
+    public String toString() {
+        String s = "";
+        for (int i = 0; i < friendCount; ++i) {
+            s = s + friends[i].toString() + "\n";
```
Comment (twentylemon) [inline source/topic3.rst:778]: `toString()` is called automatically, so it's unnecessary here. Could be information overload though, being explicit is always fine.

**Review by twentylemon — APPROVED** [review]
_makes this easy_ => 18 commits.

File: source/topic3.rst (hunk 1 above)
Comment (jameshughes89) [reply source/topic3.rst:778]: I will add this point to the aside in the string builder thing.

**Comment by jameshughes89** [comment]
The commits are because I reused the branch.

**Review by twentylemon — APPROVED** [review]

**Pull request merged**